manifests/src/update_etags.sh

```diff
@@ -1,12 +1,32 @@
 #!/usr/bin/env bash
-# Returns the updated ETag for elements in the second column of $1 alongside the first two columns.
-cat $1 |
- tail -n +2 | # Remove headers
- cut -f2 -d',' | # Select url column
- xargs -I {} -- curl -I --silent "{}" | # Fetch remote metadata 
- grep "ETag" |  # Select etag field from resulting html
- awk '{print $2}' | # Remove prefix
- sed 's/\r$//' | # Remove carriage
- sed 1i'"etag"' | # add header
- paste - $1 -d',' | # Merge with original file
- awk -F ',' '{print $2","$3","$1}' # Print in the right order
+# Fetch updated ETag values for URLs in a CSV file.
+
+input_file="$1"
+
+# Check if input file is provided
+if [ -z "${input_file}" ]; then
+  echo "Usage: $0 <input_file>"
+  exit 1
+fi
+
+url_column=$(head -n1 "${input_file}" | tr ',' '\n' | grep -n "url" | cut -d':' -f1)
+urls=$(awk -F',' -v col="${url_column}" 'NR>1 {gsub(/^"|"$/, "", $col); print $col}' "${input_file}")
+
+# Fetch ETags for each URL in a loop 
+etag_values="etag"
+while IFS= read -r url; do
+  etag=$(curl -I --silent "${url}" | awk '/[eE][tT]ag:/ {print $2}' | tr -d '\r')
```

Comment:
Regarding the pattern inside the awk one liner: We do not need to cover for multiple cases (`[eE][tT]ag`) because s3 doesn't change its etag fillename AFAIK

Reply:
fixed
```suggestion
  etag=$(curl -I --silent "${url}" | awk '/ETag:/ {print $2}' | tr -d '\r')
```